bika/lims/browser/analysisrequest/add.py

```diff
@@ -239,16 +242,16 @@ def copy_to_new_specs(self):
         copy_from = self.request.get('copy_from', "")
         if not copy_from:
             return {}
-        uids =  copy_from.split(",")
-
+        uids = copy_from.split(",")
         n = 0
         for uid in uids:
-            proxies = self.bika_catalog(UID=uid)
-            rr = proxies[0].getObject().getResultsRange()
+            proxies = self.analysisrequest_catalog(UID=uid)
+            rr = proxies[0].getObject().getResultsRange() if proxies else []
```

Comment:
As per
https://github.com/naralabs/bika.lims/blob/wip/bika/lims/content/analysis.py#L727
`getResultsRange()` returns a dict, so the following is better:

```python
rr = proxies[0].getObject().getResultsRange() if proxies else {}
```
In any case, I'd add a logger.warn here at least, cause looking for an specific object (by using its UID) and no obtaining any result is quite strange. Since the uids are retrieved from the request, it might happen the value for the uid param passed in has been manipulated manually, but... strange still. To, at least, keep track of this, better the following code:
```python
proxies = self.analysisrequest_catalog(UID=uid)
if not proxies:
    logger.warn('No object found for UID {0} while copying specs'.format(uid))
    continue
rr = proxies[0].getObject().getResultRange()
```
Also note that the whole loop could be refactored and search for all uids at once:
```python
proxies = self.analysisrequest_catalog(UID=uids)
for proxie in proxies:
    rr = proxie.getObject().getResultsRange()
    new_rr = []
    ...
    specs[n] = new_rr